packages/utils/src/features/safe-shield/hooks/address-analysis/address-activity/useAddressActivity.ts

```diff
@@ -0,0 +1,91 @@
+import { useMemo, useRef, useState } from 'react'
+import type { AsyncResult } from '@safe-global/utils/hooks/useAsync'
+import { JsonRpcProvider } from 'ethers'
+import { analyzeAddressActivity, isLowActivityAddress } from './addressActivityService'
+import { ActivityMessages } from '../config'
+import { type AnalysisResult, RecipientStatus, Severity } from '../../../types'
+import { useEffectDeepCompare, useAsyncDeepCompare } from '../../util-hooks'
+
+export type AddressActivityResult = Record<
+  string,
+  AnalysisResult<RecipientStatus.LOW_ACTIVITY | RecipientStatus.HIGH_ACTIVITY>
+>
+
+/**
+ * React hook to analyze activity for multiple addresses
+ * @param addresses - Array of Ethereum addresses to analyze
+ * @returns Object containing activity results for each address, loading state, and error
+ */
+export const useAddressActivity = (
+  addresses: string[],
+  web3ReadOnly?: JsonRpcProvider,
```

Comment:
Why do we need to pass these as props? The hook can call useOwnedSafes and useWeb3ReadOnly on its own.